simulator-spring-boot/src/main/java/org/citrusframework/simulator/http/HttpOperationScenarioRegistrar.java

```diff
@@ -0,0 +1,28 @@
+package org.citrusframework.simulator.http;
+
+import org.citrusframework.openapi.OpenApiSpecification;
+import org.citrusframework.openapi.OpenApiSpecificationProcessor;
+import org.citrusframework.simulator.config.SimulatorConfigurationProperties;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ConfigurableApplicationContext;
+
+/**
+ * Registrar for HTTP operation scenarios based on an OpenAPI specification.
+ * <p>
+ * This class implements the {@link OpenApiSpecificationProcessor} interface and processes an OpenAPI specification
+ * to register HTTP operation scenarios.
+ * </p>
+ */
+public class HttpOperationScenarioRegistrar implements OpenApiSpecificationProcessor {
+
+    @Override
+    public void process(OpenApiSpecification openApiSpecification) {
+
+        HttpScenarioGenerator generator = new HttpScenarioGenerator(openApiSpecification);
+        ApplicationContext applicationContext = SimulatorConfigurationProperties.getApplicationContext();
```

Comment:
why not just inject this? or pass it into the class using a constructor, even if this is not a bean? that static "hack" is a bit.. hacky 🤣

Reply:
Simply because instances of this class are instantiated by spi. The instantiator does not know anything about Spring.
I know it is hacky but I doubt ... no wait. I think this can be done with the Citrus instance class. That is still static but it's the Citrus way.